src/runtime.js

```diff
@@ -135,9 +135,9 @@ export function adopt(global) {
    * @param {function()=} opt_flush
    * @export
    */
-  global.AMP.setTickFunction = (fn, opt_flush) => {
+  global.AMP.setTickFunction = (fn, opt_flush, opt_setParams) => {
```

Comment:
Probably should have done this from the beginning, but could you use viewer.sendMessage() for all three of these messages to the viewer?

This is how all other communication with the viewer is done and it simpler for the viewer since it just receives whatever messages you send and we don't have to set up the callback by calling setTickFunction().
https://github.com/ampproject/amphtml/blob/master/src/service/viewer-impl.js#L703